extensions/sql-migration/src/models/migrationWizardPage.ts

```diff
@@ -6,9 +6,24 @@
 import * as azdata from 'azdata';
 import { MigrationStateModel, StateChangeEvent } from './stateMachine';
 export abstract class MigrationWizardPage {
-	constructor(protected readonly wizardPage: azdata.window.WizardPage, protected readonly migrationStateModel: MigrationStateModel) { }
+	constructor(private readonly wizard: azdata.window.Wizard, protected readonly wizardPage: azdata.window.WizardPage, protected readonly migrationStateModel: MigrationStateModel) { }
 
-	public abstract async registerWizardContent(): Promise<void>;
+	public registerWizardContent(): Promise<void> {
+		return new Promise<void>(async (resolve, reject) => {
+			this.wizardPage.registerContent(async (view) => {
+				try {
+					await this.registerContent(view);
+					resolve();
+				} catch (ex) {
+					reject(ex);
+				} finally {
```

Comment:
Not sure how this finally could ever be reached

Reply:
yeah it probably won't be, relic of another time I was putting stuff in the catch